### ISSUE TITLE
Incorrect console.log output for multiple arguments

### ISSUE THREAD

**Issue by lborgman** [firebase/firebase-tools#1817]
### [REQUIRED] Environment info
**firebase-tools:**
7.8.0

**Platform:**
Windows 10

### [REQUIRED] Test case

Under "firebase serve". (I have not deployed and tested.)

If pathIn begins with "/" this no longer finds the file:
        fileInRef = defaultBucket.file(pathIn);

### [REQUIRED] Steps to reproduce
See above.

### [REQUIRED] Expected behavior
fileInRef.exists() should return true

### [REQUIRED] Actual behavior
returns false (and that is ok, but fileInRef does not point to the existing file)

Removing the leading "/" makes the code work again. This worked (for a long time) before updating to 7.8.0

**Comment by lborgman** [comment]
There seems to be something badly broken. Maybe it is the Node.js version? This is what I see at the moment:

        const exists = await fileInRef.exists();
        console.log(exists, !exists) // false false

        const f = false;
        console.log(f, !f) // false true

**Comment by samtstern** [comment]
@lborgman can you provide a little more code?  Ideally something I could run myself and reproduce the same error.  It sounds like something is going wrong but I need to determine if this is in `firebase-tools` or maybe in `firebase-admin-node` or somewhere else.

Also if  you could show your `package.json` or at least  the `dependencies` section that would be useful as well.

**Comment by lborgman** [comment]
Now I have tested with Node.js 10.17.0 and 10.15.3
Same problem.

**Comment by lborgman** [comment]
@samtstern 

  "dependencies": {
    "@google-cloud/speech": "^3.4.0",
    "@google-cloud/storage": "^3.5.0",
    "ajv": "^6.10.2",
    "braintree": "^2.21.0",
    "busboy": "^0.3.1",
    "colors": "^1.4.0",
    "cross-spawn": "^6.0.5",
    "express": "^4.17.1",
    "fast-crc32c": "^1.0.7",
    "ffprobe-static": "^3.0.0",
    "file-system": "^2.2.2",
    "firebase": "^6.6.2",
    "firebase-admin": "^8.8.0",
    "firebase-functions": "^3.3.0",
    "formidable": "^1.2.1",
    "get-audio-duration": "^1.0.0",
    "getusermedia-to-text": "^1.0.5",
    "grpc": "^1.24.2",
    "music-metadata": "^4.9.2",
    "node-fetch": "^2.6.0",
    "request-ip": "^2.1.3",
    "serialize-error": "^4.1.0",
    "stripe": "^7.13.0",
    "websocket-stream": "^5.5.0",
    "yargs": "^14.2.2"
  },
  "devDependencies": {
    "eslint": "^6.6.0",
    "eslint-plugin-promise": "^4.2.1",
    "firebase-functions-test": "^0.1.7"
  },

**Comment by lborgman** [comment]
const functions = require('firebase-functions');
      const admin = require("firebase-admin");

      const defaultBucket = admin.storage().bucket();

      // This is were I now must remove the / => "mytmp/"
      const pathIn = "/mytmp/" + userEmail + "/" + uploadedfilename;

      const fileInRef = defaultBucket.file(pathIn);
      const exists = fileInRef.exists();

      // exists is now in the bad state I described above. 
      // It looks like it is not stored correctly since neither "exists === true" or "exists == false" is true.

**Comment by samtstern** [comment]
Ok so the relevant dependencies are:
```
    "@google-cloud/storage": "^3.5.0",
    "firebase": "^6.6.2",
    "firebase-admin": "^8.8.0",                
    "firebase-functions": "^3.3.0",
```

Let me see if I can reproduce.

**Comment by lborgman** [comment]
Something interesting, I believe. I used util.inspect:

            console.log("exists", util.inspect(exists, false, 1, true))
            //  exists [ false ]

So I tested this:

        const [exists] = await fileInRef.exists();

Now "exists" behaves as a normal boolean variable.

**Comment by samtstern** [comment]
@lborgman ah thanks!  Looks like that's the expected usage within `@google-cloud/storage`, here's an example where they do the same:
https://github.com/googleapis/nodejs-storage/blob/d77208bee82dc2d76c72fbe8d5db1bfeb8e392ff/samples/system-test/files.test.js#L61

Also you can see in their release notes that allowing leading slashes in the file name is a `4.0.0` feature:
https://github.com/googleapis/nodejs-storage/releases/tag/v4.0.0

If you have any more issues or questions please file them on the `nodejs-storage` repo and they can help you out, since it looks like this didn't turn out be related to `firebase-tools`.

**Comment by lborgman** [comment]
But ...
The variable "exists" is really strange. Is this perhaps a feature of Node.js?

When use util.inspect it is "[false]", but when using just console.log(exists) it is "false", but behaves very strange, see above.

Can you confirm that Node.js + Firebase serve should behaves this way? Or please reopen this issue.

**Comment by lborgman** [comment]
Maybe the problem is just in console.log under "firebase serve"?

        const existsStrange = await fileInRef.exists();
        console.log("existsStrange", typeof existsStrange, existsStrange)
        // existsStrange object false

**Comment by samtstern** [comment]
@lborgman hey you're right there's an issue with our `console.log()`.

In a plain `node` REPL:

```
$ node
> const x = [false];
> console.log("strange", typeof x, x)
strange object [ false ]
```

In `firebase serve`:

```
>  strange object false
```

Thanks for pointing this out!  Let's repurpose this bug to track that fix.

**Comment by samtstern** [comment]
This was fixed by #1824 but never closed.
